File: Data Structures & Algorithms/longest-common-prefix/submission-2.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        output = ""
        for i, char in enumerate(strs[0]):
            for val in strs[1:]:
                if i >= len(val) or char != val[i]:
                    return output
            output += char
        return output

File: Data Structures & Algorithms/anagram-groups/submission-2.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> result = new ArrayList<>();
        Map<String, List<String>> subStrs = new HashMap<>();

        for(String str : strs) {
            char[] chars = str.toCharArray();
            Arrays.sort(chars);
            String sortedStr = new String(chars);

            if(subStrs.get(sortedStr) == null) subStrs.put(sortedStr, new ArrayList<String>());
            subStrs.get(sortedStr).add(str);
        }

        for(String key : subStrs.keySet()) result.add(subStrs.get(key));
        return result; 
    }
}
